### PR TITLE
feat(tds-table): added horizontal scroll

### PR DESCRIPTION
**Pull request by Lunkan89**
## **Describe pull-request**  
_Describe what the pull-request is about_

## **Issue Linking:**  
_Choose one of the following options_
- **Jira:** Add ticket number after `CDEP-`: [[CDEP-](https://tegel.atlassian.net/browse/CDEP-)    ](https://tegel.atlassian.net/browse/CDEP-3368)
- **GitHub:** Include issue link  
- **No issue:** Describe the problem being solved.  

## **How to test**  
_Provide detailed steps for testing, including any necessary setup._
1. Go to...
2. Check in...
3. Run ...

## **Checklist before submission**
- [ ] I have added unit tests for my changes (if applicable)
- [X] All existing tests pass
- [ ] I have updated the documentation (if applicable)

## **Suggested test steps**
- [ ] Browser testing (Chrome, Safari, Firefox) 
- [ ] Keyboard operability
- [ ] Interactive elements have labels.
- [ ] Storybook controls
- [ ] Design/controls/props is aligned with other components 
- [ ] Dark/light mode and variants 
- [ ] Input fields – values should be displayed properly 
- [ ] Events

## **Screenshots**  
_Include before/after screenshots for UI changes._

## **Additional context**  
_Add any other context or feedback requests about the pull-request here._

### CODE REVIEW

File: packages/core/src/components/table/table-component-basic.stories.tsx [hunk 1]
```diff
@@ -109,6 +109,31 @@ export default {
       control: {
         type: 'boolean',
       },
+      table: {
+        defaultValue: { summary: false },
+      },
+    },
+    horizontalScroll: {
+      name: 'Horizontall scrolling',
```
Comment (timrombergjakobsson) [inline packages/core/src/components/table/table-component-basic.stories.tsx:117]:
```suggestion
      name: 'Horizontal scrolling',
```

File: packages/core/src/components/table/table/table.scss [hunk 2]
```diff
@@ -11,3 +11,19 @@
...
   width: 100%;
 }
+
+.tds-table--horizontall-scroll {
+  display: block;
+  overflow-x: scroll;
+  white-space: nowrap;
+}
+
+::-webkit-scrollbar {
```
Comment (timrombergjakobsson) [inline packages/core/src/components/table/table/table.scss:21]: Just a thought, we have a `_scrollbar.scss`
https://github.com/scania-digital-design-system/tegel/blob/a1b37a6f411b5d896e9883a2460f86c90a59e187/packages/core/src/mixins/_scrollbar.scss#L2

that just supports vertical scroll. Maybe we could build upon it and create a horizontal one also?
Something like:
`@mixin tds-scrollbar($orientation: vertical) { }`

File: packages/core/src/components/table/table/table.tsx [hunk 3]
```diff
@@ -63,6 +63,12 @@ export class TdsTable {
   /** Variant of the component, based on current mode. */
   @Prop({ reflect: true }) modeVariant: 'primary' | 'secondary' = null;
 
+  /** Enabling horizontal scroll */
+  @Prop() horizontalScroll: boolean = false;
+
+  /** Width of the used as the constraint width in conjunction with horizontal scrolling  */
```
Comment (timrombergjakobsson) [inline packages/core/src/components/table/table/table.tsx:69]: Do you mean?
```suggestion
  /** Width of the table, used as the constraint for horizontal scrolling. */
```

**Comment by timrombergjakobsson** [comment]
Maybe design should have a look at this as well? @charlottelorensson

File: packages/core/src/components/table/table/table.scss [hunk 4]
```diff
@@ -11,3 +12,10 @@
 .tds-table--responsive {
   width: 100%;
 }
+
+.tds-table--horizontall-scroll {
```
Comment (nathalielindqvist) [inline packages/core/src/components/table/table/table.scss:16]:
```suggestion
.tds-table--horizontal-scroll {
```

File: packages/core/src/components/table/table/table.tsx [hunk 5]
```diff
@@ -159,12 +165,14 @@ export class TdsTable {
...
       >
         <table
+          style={this.horizontalScroll ? { width: this.width } : {}}
           class={{
             'tds-table': true,
             'tds-table--compact': this.compactDesign,
             'tds-table--divider': this.verticalDividers,
             'tds-table--no-min-width': this.noMinWidth,
             'tds-table--responsive': this.responsive,
+            'tds-table--horizontall-scroll': this.horizontalScroll,
```
Comment (nathalielindqvist) [inline packages/core/src/components/table/table/table.tsx:175]:
```suggestion
            'tds-table--horizontal-scroll': this.horizontalScroll,
```

**Review by nathalielindqvist — APPROVED** [review]
Apart from the small comments in the class naming I think this looks good! Nice job! 🚀

**Comment by Lunkan89** [comment]
After discussion with @theJohnnyMe and Laurens improvements we wanted to do was:
1. Only use one prop for activating horizontal scrolling ( `horizontalScrollWidth` is used now )
2. Have both footer and tooltip (if in the table) to be "sticky"
3. Create a separate story-test for horizontal scrolling to be able to fully test everything
4. Also being able to set column width when horizontal scroll i activated, not just when setting `noMinWidth` flag

Note:
`noMinWidth` should still work to set width of columns, although i cannot use multiple conditionals for showing column width input on both horizontal scroll flag and minWidth flag. https://github.com/storybookjs/storybook/discussions/18542

**Comment by Lunkan89** [comment]
The story includes both pagination and table tooltip, so there is some code duplication there.
Any suggested improvements?

**Comment by Lunkan89** [comment]
Accepted by design :)

**Review by timrombergjakobsson — APPROVED** [review]
Nice work @Lunkan89!

**Pull request merged**